### DISCUSSION TITLE
How to update the position of a vertexs

### DISCUSSION THREAD

**Discussion by yishanxiansheng** [google/filament#6225] [Q&A] [answered]
I have parsed the positions of the mesh vertices, when I change the position of one of the points, how can I make it works?

I noticed that when changing the position value of the vertex during the loading stage（ResourceLoader -> loadResources()）, the position of the vertex does move. However, when updating the position value of the vertex during update stage (), there is no movement.

**Comment by romainguy** [comment, marked as answer]
You need to reupload the modified vertices in the VertexBuffer. We have APIs for this.

**Reply by binzi56** [reply]
Help me look at this code, I can update the vertex data, but there is a memory leak, the memory continues to grow, I don't know where the problem is; I guess it may be that the BufferObject uses the cache and may not be released, I don't know how to deal with it?

```c++
void MeshHelper::uploadBufferToGPU(uint8_t *data, uint32_t size, Engine *engine) {
    BufferObject *bo = BufferObject::Builder().size(size).build(*engine);
    std::vector<filament::BufferObject*> bufferObjVec = asset->getBufferObject();

    bufferObjVec.push_back(bo);

    using BufferDescriptor = filament::backend::BufferDescriptor;
    auto *fFilamentAsset = static_cast<FFilamentAsset *>(asset);

    VertexBuffer::BufferDescriptor bd(data, size,
                     gltfio::uploadCallback, gltfio::uploadUserdata(fFilamentAsset));

    bo->setBuffer(*engine, std::move(bd));

    fFilamentAsset->getVertexBuffers()[0]->setBufferObjectAt(*engine, 0, bo);
}
```

**Reply by romainguy** [reply]
You are creating a new buffer object every time. Instead you should take the existing buffer objects and replace the data they hold.

**Reply by binzi56** [reply]
Solved, thanks for the pointer
